swag/swag_client.py

```diff
@@ -369,3 +416,335 @@ async def execute_swagdmin_command(self, message):
+                        "".join(
+                            argent
+                            for argent in splited_command
+                            if argent.replace(".", "").replace(",", "").isnumeric()
+                        )
+                    ).quantize(Decimal(".0001"), rounding=ROUND_DOWN)
+                except ValueError:
+                    raise InvalidStyleValue
+
+            self.swag_bank.payer_a_cagnotte(message.author.id, cagnotte_idx, value)
+
+            cagnotte_name = (
+                self.swag_bank.get_active_cagnotte(cagnotte_idx)
+                .get_info()
+                .cagnotte_name
+            )
+            await message.channel.send(
+                "Transaction effectué avec succès ! \n"
+                "```ini\n"
+                f"[{message.author.display_name}\t"
+                f"{format_number(value)} {currency}\t"
+                f"-->\t€{cagnotte_idx} {cagnotte_name}]\n"
+                "```"
+            )
+            await update_forbes_classement(message.guild, self, self.client)
+
+        elif "donner" in splited_command:
+            cagnotte_idx = get_cagnotte_idx_from_command(splited_command)
+
+            destinataire = message.mentions
+            if len(destinataire) != 1:
+                await message.channel.send(
+                    "Merci de mentionner un destinataire (@Bobby Machin) pour "
+                    "lui donner une partie de cette €agnotte !"
+                )
+                return
+            destinataire = destinataire[0]
+
+            currency = (
+                self.swag_bank.get_active_cagnotte(cagnotte_idx)
+                .get_info()
+                .cagnotte_currency
+            )
+
+            if currency == "$wag":
+                try:
+                    value = int(
+                        "".join(
+                            argent for argent in splited_command if argent.isnumeric()
+                        )
+                    )
+                except ValueError:
+                    raise InvalidSwagValue
+
+            elif currency == "$tyle":
+                try:
+                    value = Decimal(
+                        "".join(
+                            argent
+                            for argent in splited_command
+                            if argent.replace(".", "").replace(",", "").isnumeric()
+                        )
+                    ).quantize(Decimal(".0001"), rounding=ROUND_DOWN)
+                except ValueError:
+                    raise InvalidStyleValue
+
+            self.swag_bank.donner_depuis_cagnotte(
+                cagnotte_idx, destinataire.id, value, message.author.id
+            )
+
+            cagnotte_name = (
+                self.swag_bank.get_active_cagnotte(cagnotte_idx)
+                .get_info()
+                .cagnotte_name
+            )
+            await message.channel.send(
+                "Transaction effectué avec succès ! \n"
```

Comment:
effectuée

Reply:
Resolved in https://github.com/BobbySoftwares/Bybbot/pull/7/commits/173d6bc07cf9a1a3121d5820f16d9bcb19a74a16